modules/tinymce/src/themes/silver/main/ts/ui/core/complex/StyleFormat.ts

```diff
@@ -81,10 +81,11 @@ const mapFormats = (userFormats: AllowedFormat[]): CustomFormatMapping => Arr.fo
     };
   } else if (isInlineFormat(fmt) || isBlockFormat(fmt) || isSelectorFormat(fmt)) {
     // Convert the format to a reference and add the original to the custom formats to be registered
-    const formatName = `custom-${fmt.title.toLowerCase()}`;
+    const formatName = Type.isString(fmt.name) ? fmt.name : fmt.title.toLowerCase();
+    const formatNameWithPrefix = `custom-${formatName}`;
```

Comment:
Would it be misleading to the people using this for to specify the name as `my-format` and then have the actual name used be `custom-my-format`?

Reply:
Hmm yeah I think it's a bit weird to prefix this. But then again they are prefixed with custom and suffixes with the title as lowercase today. The important part for rtc is that they are unique and the same if you say change the title so the format name matters less and it would only matter if the user manually calls say formatter.applyFormat something I guess is not that common to do. What do you think @lnewson you suggested the change. I'm not sure both seems valid options.

Reply:
Yeah, I'm unsure as well. I largely suggested it so as to avoid conflicts, as it's a bit of a different type of format and people can declare it today without it conflicting (it also uses the custom prefix today as Spocke mentioned).

I'd also agree people don't normally attempt to use the formatter API with style formats.

Reply:
I'm happy to do whatever others think is best.

Reply:
We could also check if the format exists and blow up with an error message if it exists.

Reply:
But it would ha e to be an console.error since it's it's in the middle of the init process so it can't throw.

Reply:
Yeah, that probably makes more sense. We'd then just exclude the format in the menus then as well I'd expect?

Reply:
Argh looking at the code that would be way more complex since removing an item from the list would need everything to become bind instead of map since the item may or may not exist when registering. So I would have to change a lot more code than just this I don't think it's worth it since the name there is some what of an edge case. Will probably only require the name on inline formats for rtc since that is the only one that is affected by this.

Reply:
Ahh yeah, that makes sense. I'm honestly happy to go with whatever you think is best here Spocke.

Reply:
Ok lets just keep custom- prefix it would be a bit more problematic in the docs but the docs are pretty bad anyway for style formats I will try do make a PR where these things are more clear.